frontend/views/components/GroupSettings/InviteLinks.vue

```diff
@@ -109,72 +115,95 @@ export default {
-        dummyInviteList: [
-          {
-            name: 'Felix Kubin',
-            inviteLink: 'http://localhost:8000/app/join?groupId=21XWnNFz7RbNPKHUqAeSLLT1cNHnnCssmSw6dJeB1gfSSeZc7v&secret=4460',
-            isAnyoneLink: false,
-            state: {
-              description: 'Not used yet',
-              expireInfo: '1d 2h 30m left',
-              isExpired: false
-            }
-          },
-          {
-            name: 'Brian Eno',
-            inviteLink: 'http://localhost:8000/app/join?groupId=30aFnTYz7RbqAeSLLT1cNfSSeZHN6KHUnnCssmSw6dJeB1gc7v&secret=2250',
-            isAnyoneLink: false,
-            state: {
-              description: 'Used',
-              expireInfo: '',
-              isExpired: true
-            }
-          },
-          {
-            name: 'Carl Sagan',
-            inviteLink: 'http://localhost:8000/app/join?groupId=B1gfSSeZc721XWnNFz7RbkoyHUqAeSwLT1cNHnnCssmSw6dJev&secret=1348',
-            isAnyoneLink: false,
-            state: {
-              description: 'Not used',
-              expireInfo: 'Expired',
-              isExpired: true
-            }
-          },
-          {
-            name: 'Anyone',
-            inviteLink: 'http://localhost:8000/app/join?groupId=s8LT1cNHnnCs21XWnNFz7RbNPKHUqAesmSw6dJeB1gfSSeZc7v&secret=5521',
-            isAnyoneLink: true,
-            state: {
-              description: '10/60 used',
-              expireInfo: 'Expired',
-              isExpired: true
-            }
-          }
-        ]
+        }
       }
     }
   },
   methods: {
     unfocusSelect () {
       this.$refs.select.blur()
     },
-    toInvite (e) {
-      if (e.target.tagName === 'SPAN') this.$router.push({ path: '/invite' })
-    },
     activateWebShare (inviteLink) {
       if (navigator.share) {
         navigator.share({
-          title: this.L('Your invite'),
+          title: L('Your invite'),
           url: inviteLink
         })
       }
+    },
+    inviteStatusDescription ({
+      isAnyoneLink,
+      isInviteExpired,
+      isAllInviteUsed,
+      quantity,
+      numberOfResponses
+    }) {
+      if (isAnyoneLink) return L('{numberOfResponses}/{quantity} used', { numberOfResponses, quantity })
+      else if (isAllInviteUsed) return L('Used')
+      else return isInviteExpired ? L('Not used') : L('Not used yet')
+    },
+    readableExpiryInfo (expiryTime) {
+      const MIL = 1000
+      const MIL_MIN = 60 * MIL
+      const MIL_HR = MIL_MIN * 60
+      const MIL_DAY = 24 * MIL_HR
+      let remainder
+
+      const days = Math.floor(expiryTime / MIL_DAY)
+      remainder = expiryTime % MIL_DAY
+      const hours = Math.floor(remainder / MIL_HR)
+      remainder = remainder % MIL_HR
+      const minutes = Math.ceil(remainder / MIL_MIN)
+      const args = {
+        days: days ? `${days}d ` : '',
+        hours: hours ? `${hours}h ` : '',
+        minutes: minutes ? `${minutes}m ` : ''
+      }
+
+      return L('{days}{hours}{minutes}left', args)
```

Comment:
OK, so we are almost perfect here. :-)

Remember that in another language days, hours, and minutes likely begin with symbols other than `d`, `h`, and `m`, and that translators might want to translate that too.

So this is almost perfect, but we still need to deal with that.

I see two approaches:

### One: Have everything, including "Expired" here

```js
  if (days) return L('{days}d {hours}h {minutes}m left', { days, hours, minutes })
  if (hours) return L('{hours}h {minutes}m left', { hours, minutes })
  if (minutes) return L('{minutes}m left', { minutes })
  return L('Expired')
```

In terms of raw speed, this approach should be just as fast as what you have above with the `?:` operator, because that operator creates [a branch](https://en.wikipedia.org/wiki/Branch_(computer_science)), just like an `if` statement.

For info on why branching can slow down execution, you might want to watch some videos on ["CPU pipelining"](https://www.youtube.com/watch?v=doJpguZFTe0) and "branch prediction".

But what I'm saying is that it seems like in this case there's no way to avoid it — AND it's not really an issue in terms of performance. It only becomes an issue when:

- Giant if/else blocks are used where they're unnecessary, making code ugly and hard to read
- Giant if/else blocks are located in a performance "hot zone", i.e. in code that gets executed constantly, such as in the main logic of a video game's run-loop

My personal vote is to go with this approach shown above, since it includes the "Expired" string within this function, and because it generates strings that give more context to translators.

The other approach is a modified version of what you have above.

### Two: Modified version of above

```js
const args = {
  days: days ? L('{numDays}d', { numDays: days }) : '',
  hours: hours ? L('{numHours}h', { numHours: hours }) : '',
  minutes: minutes ? L('{numMinutes}m', { numMinutes: minutes }) : '',
}
// HTML converts multiple spaces into a single space (unless &nbsp; is used)
return L('{days} {hours} {minutes} left', args)
```

Reply:
Thanks for the mini compsci session here! 👍